src/olmo_core/model_ladder2/wsds_chinchilla_run_configurator.py

```diff
@@ -0,0 +1,223 @@
+import dataclasses
+import math
+from dataclasses import dataclass
+
+from olmo_core.aliases import PathOrStr
+from olmo_core.exceptions import OLMoConfigurationError
+from olmo_core.optim import (
+    WSDS,
+    OptimGroupOverride,
+    Scheduler,
+    SchedulerUnits,
+    SkipStepAdamWConfig,
+)
+from olmo_core.train import Duration
+
+from .base import RunConfigurator
+from .utils import format_count, format_tokens
+
+
+@dataclass(kw_only=True)
+class WSDSChinchillaRunConfigurator(RunConfigurator):
+    """
+    A run configurator that uses WSD-S learning rate scheduling and Chinchilla scaling laws.
+
+    .. note::
+        You may need to tune the :data:`tokens_per_param` value to your dataset and optimizer.
+    """
+
+    chinchilla_multiple: float
+    """
+    How long to train each run for, expressed as a multiple of the Chinchilla-optimal duration
+    which must be a power of 2.
+    """
+    decay_fraction: float = 0.1
+    """The duration of each decay as a fraction of the period. Must be at least 10%."""
+    tokens_per_param: int = 20
+    """The number of tokens per parameter to use for Chinchilla calculations."""
+
+    def __post_init__(self):
+        if self.chinchilla_multiple < 0.5 or not math.log(self.chinchilla_multiple, 2).is_integer():
+            raise OLMoConfigurationError(
+                "'chinchilla_multiple' must be at least 0.5 and a power of 2"
+            )
+        if not (0 < self.decay_fraction < 0.5):
+            raise OLMoConfigurationError(
+                "'decay_fraction' must be greater than 0.0 and less than 0.5"
+            )
+
+    def configure_duration(self, num_params: int) -> Duration:
+        return self._chinchilla_duration(
+            num_params,
+            self.chinchilla_multiple,
+        )
+
+    def configure_target_batch_size(self, num_params: int) -> int:
+        # Calculate global batch size according to https://api.semanticscholar.org/CorpusID:270764838
+        # which assumes a sequence length of 2048.
```

Comment:
hmmm, any idea if / how this would be modified for 8k seq len?

Reply:
I'm not sure how brittle these scaling equations are to sequence length. We're assuming that 8k is not too different from 2k, so the target batch size in tokens is not a function of sequence length. That seems reasonable enough.